### ISSUE TITLE
ctrl+/ cannot switch between source code mode and read mode.

### ISSUE THREAD

**Issue by superPershing** [typora/typora-issues#753]
I'm a Typora user under ubuntu16.04.2 LTS. When I write in Typora, I cannot use hotkey `ctrl+/` to switch between source code mode and read mode. I had to switch two mode by mouse, it was terrible and wasting time. But it was effective when I use windows before. What should I do? Or there are some bugs on the version of linux?

**Comment by abnerlee** [comment]
The default shortcut key is `ctrl+/`

**Comment by superPershing** [comment]
sorry，it is my fault, I want to type ctrl+/ and what I mean is ctrl + /. In my Ubuntu, I cannot switch between source code mode and read mode by ctrl + /.

**Comment by abnerlee** [comment]
Are you using English-US keyboard layout? and did you use any IME?

**Comment by superPershing** [comment]
@abnerlee I use sougou input. When I close sougou input or switch to English-US keyboard, It occur the same error as before.

**Comment by superPershing** [comment]
@abnerlee And When I in the read mode, type `ctrl + /` will select all the text, look like `ctrl + A`.

**Comment by abnerlee** [comment]
I see, so for other apps, `ctrl+/` will not select all text like `ctrl+A`, right?

And `apt-get remove`, then install typora again, will not fix the bug?

I will check on my Ubuntu later.

**Comment by Cooper-Yang** [comment]
Have the same problem. Also using sogou input in Ubuntu 16.04.2.

However, in my case, `ctrl + /` work as `ctrl + a` globally, still trying to figure it out.

@abnerlee Any suggestion for solving this?

**Comment by abnerlee** [comment]
> However, in my case, ctrl + / work as ctrl + a globally

@Cooper-Yang, does `globally` means for other apps, ctrl+/ also work as ctrl+a?

(For example, this textarea in Github issue page)

**Comment by Cooper-Yang** [comment]
@abnerlee Yes.

**Comment by abnerlee** [comment]
If so, then it is not Typora's bug, but sougou IME's.

Please try 
- Assign Typora's `switch to source code mode` action to other shortcut keys (http://support.typora.io/Custom-Key-Binding/)
- Open the configuration of sougou or ibus or Fcitx, and to check if `ctrl+/` is binded as shortcut key for some operation (like select all)
- since it can be reproduced in other apps, please try update sougou or contact with sougou (for example: http://pinyin.sogou.com/bbs/forum.php) for help

**Comment by Cooper-Yang** [comment]
@abnerlee That solution might not work, since the "Open Advanced Settings" does nothing in Ubuntu. Check #642 . Appreciate if you can kindly help me locate the config file.

TL;DR: I know that its not typora's fault, but after purging sogou and fcitx and reboot, It is clear that they cannot be blame either, as `ctrl+/` still work as select all globally.

**Comment by abnerlee** [comment]
Sorry, 

First click `Reset Advanced Settings`, then try open "Open Advanced Settings" again.

So maybe `ctrl+/` is assigned as `Select All` on Ubuntu? maybe this doc (https://help.ubuntu.com/community/KeyboardShortcuts) can help for finding about whether `ctrl+/` is a shortcut key on your ubuntu.

I remember my Ubuntu dose not have this problem, but I will confirm again later.

**Comment by Cooper-Yang** [comment]
> First click Reset Advanced Settings, then try open "Open Advanced Settings" again.

Thx, It work perfectly :-)

**Comment by LeBaux** [comment]
I ended up here searching for "Toggle Source Code Mode" shortcut `ctrl+/` and there is not much to be found so I will elaborate here as I have similar issue. 

Shortcut `ctrl+/` is not working with most of the Europe keyboard layouts. It is also not working if you press `/` on numpad.  Since I often write in different languages, it would be beneficial if the shortcut worked with other layouts.

So I've tried mapping the shortcut on different keys. Edited `conf.user.json`, adding just one line before last bracket, `"keyBinding": {"Toggle Source Code Mode": "Ctrl+Shift+M"},`. I've tried multiple shortcuts to no success. What am I doing wrong? Thanks for help!

**Comment by superPershing** [comment]
@LeBaux Try removing `Toggle`? Like `"Source Code Mode": "Ctrl+Shift+M",`

**Comment by LeBaux** [comment]
@superPershing that did the trick, thank you for your help! 🙏 For anyone landing here from google, if you want to customize "Toggle Source Code Mode" shortcut:

1. Open Typora, go to FILE > PREFERENCES > OPEN ADVANCED SETTINGS and 
2. Open file `conf.user.json`, and add `,"keyBinding": {"Toggle Source Code Mode": "Ctrl+Shift+M"}` in front of last bracket (`}`).
3. Save, restart Typora. You might want to use different binding, I use "Ctrl+Space".

Note that this might change since the app is under active development.

**Comment by jincera** [comment]
In my computer, it's Ctrl + } even if in the menu it says Ctrl + /

**Comment by pablogzlezmora** [comment]
In my computer with Mageia it's CTRL + Ç, Spanish keyboard (for "Ç" key, "}" is the ALT GR input).

**Comment by abnerlee** [comment]
@pablogzlezmora  please consider reassign a shortcut key http://support.typora.io/Shortcut-Keys/#change-shortcut-keys

**Comment by pablogzlezmora** [comment]
Thanks, @abnerlee.

**Comment by sorsater** [comment]
> @superPershing that did the trick, thank you for your help! For anyone landing here from google, if you want to customize "Toggle Source Code Mode" shortcut:
> 
> 1. Open Typora, go to FILE > PREFERENCES > OPEN ADVANCED SETTINGS and
> 2. Open file `conf.user.json`, and add `,"keyBinding": {"Toggle Source Code Mode": "Ctrl+Shift+M"}` in front of last bracket (`}`).
> 3. Save, restart Typora. You might want to use different binding, I use "Ctrl+Space".
> 
> Note that this might change since the app is under active development.
> If you want to add multiple key bindings, you can use this format:
> ` ,"keyBinding": {"File Tree": "Shift+Space", "Source Code Mode": "Ctrl+Space"}`

Note, For me it worked with "keyBinding": {"Source Code Mode": "Ctrl+Shift+M"}, so no "Toggle"

**Comment by pablogzlezmora** [comment]
Thanks, @sorsater
